pkg/registry/batch/cronjob/storage/storage.go

```diff
@@ -84,3 +101,49 @@ func (r *StatusREST) Get(ctx genericapirequest.Context, name string, options *me
 func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation)
 }
+
+// InstantiateREST implements the REST endpoint for manually triggering a CronJob.
+type InstantiateREST struct {
+	cronJobStore *genericregistry.Store
+	jobStore     *genericregistry.Store
+}
+
+var _ = rest.NamedCreater(&InstantiateREST{})
+
+func (r *InstantiateREST) New() runtime.Object {
+	return &batch.CronJobManualInstantiation{}
+}
+
+func (r *InstantiateREST) Create(ctx genericapirequest.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, includeUninitialized bool) (out runtime.Object, err error) {
+	sourceCronJobObj, err := r.cronJobStore.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	sourceCronJob, ok := sourceCronJobObj.(*batch.CronJob)
+	if !ok {
+		return nil, fmt.Errorf("got object of type %T, expected *batch.CronJob", sourceCronJobObj)
+	}
+
+	job := &batch.Job{
```

Comment:
Make [getJobFromTemplate](https://github.com/kubernetes/kubernetes/blob/2cbb07a4394ef4cfc1666e114674a047023ed178/pkg/controller/cronjob/utils.go#L173-L194) public and move it here, so that storage (which is part of API server) does not depend on controller, that's not good.